Entity/CategoryManager.php

```diff
@@ -143,11 +143,11 @@ public function getRootCategory($context = null)
     /**
      * NEXT_MAJOR: add this method to the interface.
      *
-     * @param ContextInterface $context
+     * @param ContextInterface|string|null $context
      *
      * @return CategoryInterface[]
      */
-    public function getRootCategoriesForContext(ContextInterface $context = null)
+    public function getRootCategoriesForContext($context = null)
```

Comment:
Then I think it's safe to turn a blind eye to the BC-break and make the method final. This will help us reason safely in the future (for instance, when adding return type hinting for php 7)